### PR TITLE
vm support: don't generate virtual inbound/outbound listeners for proxies that are not using Iptables redirection

### PR DESCRIPTION
**Pull request by yskopets**
Signed-off-by: Yaroslav Skopets <yaroslav@tetrate.io>

**Please provide a description of this PR:**

This PR turns off generation of virtual inbound/outbound listeners for proxies that are not using Iptables redirection.

The purpose of this PR is to remove configuration Istio sidecar doesn't need, minimize potential for conflicts over ports, get closer to the support of multiple Istio Sidecars per VM.

### CODE REVIEW

**Comment by howardjohn** [comment]
Seems reasonable to me but i don't know much about this mode. Assigned some experts

File: pilot/pkg/networking/core/v1alpha3/listener_builder.go [hunk 1]
```diff
@@ -350,6 +350,12 @@ func (lb *ListenerBuilder) buildHTTPProxyListener(configgen *ConfigGeneratorImpl
 }
 
 func (lb *ListenerBuilder) buildVirtualOutboundListener(configgen *ConfigGeneratorImpl) *ListenerBuilder {
+	noneMode := lb.node.GetInterceptionMode() == model.InterceptionNone
+	if noneMode {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/listener_builder.go:354]: Nit: the two lines can be merged 

Except for this looks good to me.

Comment (yskopets) [reply pilot/pkg/networking/core/v1alpha3/listener_builder.go]: merged

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**